go.mod

```diff
@@ -473,17 +473,32 @@ require (
 require github.com/jmespath-community/go-jmespath v1.1.1 // @grafana/identity-access-team
 
 require (
-	cloud.google.com/go/longrunning v0.5.12 // indirect
+	cloud.google.com/go/longrunning v0.6.0 // indirect
 	github.com/at-wat/mqtt-go v0.19.4 // indirect
+	github.com/c2h5oh/datasize v0.0.0-20231215233829-aa82cc1e6500 // indirect
+	github.com/cespare/xxhash v1.1.0 // indirect
 	github.com/dolthub/maphash v0.1.0 // indirect
 	github.com/emirpasic/gods v1.18.1 // indirect
 	github.com/gammazero/deque v0.2.1 // indirect
 	github.com/grafana/grafana-app-sdk v0.19.0 // indirect
 	github.com/grafana/grafana/pkg/semconv v0.0.0-20240808213237-f4d2e064f435 // indirect
+	github.com/grafana/jsonparser v0.0.0-20240425183733-ea80629e1a32 // indirect
+	github.com/grafana/loki/pkg/push v0.0.0-20231124142027-e52380921608 // indirect
+	github.com/grafana/loki/v3 v3.2.1 // indirect
```

Comment:
This is relatively heavy as it will need "whole" Loki to work. Just out of curiosity: scopes is purely backend? So validation can't happen on the frontend?

Reply:
So far scopes are only injected in the backend. 
We wanted to use the promql parser when injecting labels into prometheus queries. If there is a logql parser in javascript we would entertain using that instead but I rather stick to doing it in the backend for consistency.

Reply:
> If there is a logql parser in javascript we would entertain using that instead but I rather stick to doing it in the backend for consistency.

yea, makes sense. I know we removed `github.com/grafana/loki` from our go dependencies at one point. And also there were some discussions if the Loki team could extract their LogQL syntax parser into a separate package. I think that might make sense in the long run.

Reply:
>  Loki team could extract their LogQL syntax parser into a separate package
I was about to do just that 

It's a big one, but one we need to do once and for all. I'm on it.

Reply:
I'm almost done I think if that is of any interest